projects/fsastorefrontlib/src/core/my-account/services/policy-chart-data.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+import { EChartsOption } from 'echarts';
+
+@Injectable()
+export class PolicyChartDataService {
+  protected readonly SEPARATOR = '.';
+
+  groupPoliciesByAttribute(policies: any[], key: string | string[]) {
+    return policies.reduce((policy, item) => {
+      const groupCriteria = this.getObjectValueByProperty(key, item);
+      policy[groupCriteria] = policy[groupCriteria] || [];
+      policy[groupCriteria].push({
+        categoryData: item.categoryData,
+        policyPremium: item.policyPremium,
+        paymentFrequency: item.paymentFrequency,
+      });
+      return policy;
+    }, {});
+  }
+
+  calculatePremiumAmountByCategory(policies, chartOption: EChartsOption) {
+    chartOption.series[0].data = [];
+    Object.keys(policies).forEach(category => {
+      const reducer = (sum, policy) => sum + policy.policyPremium.value;
+      const initialValue = 0;
+      const policyAmountByCategory = policies[category].reduce(
+        reducer,
+        initialValue
+      );
```

Comment:
No need for over-complication:
```      
const policyAmountByCategory = policies[category].reduce(
        (sum, policy) => sum + policy.policyPremium.value,
        0
      );
```

Reply:
I separate that only for better readability, not for complication :)

Reply:
I would agree, but also I wouldn't say `policies[category]` is hard to read, and putting reduce function in separate constant is okay when that function has a lot of logic. Here it only does the default reduction.